### PR TITLE
[stdlib] Additional ownership annotations

### PR DESCRIPTION
**Pull request by airspeedswift**
What it says...

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please smoke benchmark

**Comment by airspeedswift** [comment]
@swift-ci please smoke test

**Comment by airspeedswift** [comment]
@swift-ci please benchmark

File: stdlib/public/core/Array.swift [hunk 1]
```diff
@@ -1490,7 +1490,7 @@ extension Array {
   }
 
   @inlinable
-  public func _copyContents(
+  __consuming public func _copyContents(
```
Comment (xwu) [inline stdlib/public/core/Array.swift]: `public __consuming`?

Comment (airspeedswift) [reply stdlib/public/core/Array.swift]: gah, thx

**Comment by airspeedswift** [comment]
@swift-ci please smoke test and merge

**Comment by airspeedswift** [comment]
@swift-ci please smoke test and merge

**Pull request merged**